includes/API.php

```diff
@@ -49,6 +48,52 @@ public function __construct( $access_token ) {
 	}
 
 
+	/**
+	 * Validates a response after it has been parsed and instantiated.
+	 *
+	 * Throws an exception if a rate limit or general API error is included in the response.
+	 *
+	 * @since 2.0.0-dev.1
+	 *
+	 * @throws Framework\SV_WC_API_Exception
+	 */
+	protected function do_post_parse_response_validation() {
+
+		/** @var API\Response $response */
+		$response = $this->get_response();
+
+		if ( $response && $response->has_api_error() ) {
+
+			$message = sprintf( '%s: %s', $response->get_api_error_type(), $response->get_api_error_message() );
+
+			/**
+			 * Graph API
+			 *
+			 * 4 - API Too Many Calls
+			 * 17 - API User Too Many Calls
+			 * 32 - Page-level throttling
+			 * 613 - Custom-level throttling
+			 *
+			 * Marketing API (Catalog Batch API)
+			 *
+			 * 80004 - There have been too many calls to this ad-account
+			 *
+			 * @link https://developers.facebook.com/docs/graph-api/using-graph-api/error-handling#errorcodes
+			 * @link https://developers.facebook.com/docs/graph-api/using-graph-api/error-handling#rate-limiting-error-codes
+			 * @link https://developers.facebook.com/docs/marketing-api/reference/product-catalog/batch/#validation-rules
+			 */
+			if ( in_array( $response->get_api_error_code(), [ 4, 17, 32, 613, 80004 ], true ) ) {
+
+				throw new API\Exceptions\Request_Limit_Reached( $message, $response->get_api_error_code() );
+
+			} else {
```

Comment:
`else` not needed since the first throw would stop the train and the latter throw would be executed if the first if condition is false

Reply:
Good point. I removed the `else` in e3b2472.